core/src/autogluon/core/searcher/bayesopt/autogluon/constrained_gp_fifo_searcher.py

```diff
@@ -0,0 +1,262 @@
+                 local_minimizer_class: Type[LocalOptimizer] = DEFAULT_LOCAL_OPTIMIZER_CLASS,
+                 skip_optimization: SkipOptimizationPredicate = None,
+                 num_initial_candidates: int = DEFAULT_NUM_INITIAL_CANDIDATES,
+                 num_initial_random_choices: int = DEFAULT_NUM_INITIAL_RANDOM_EVALUATIONS,
+                 initial_scoring: Optional[str] = None,
+                 first_is_default: bool = True,
+                 debug_log: Optional[DebugLogPrinter] = None,
+                 cost_metric_name: Optional[str] = None,
+                 profiler: Optional[SimpleProfiler] = None,
+                 getconfig_callback: Optional[Callable[[dict], Any]] = None):
+        """
+        Note that the SurrogateModel is created on demand (by the state
+        transformer) in get_config, along with components needed for the BO
+        algorithm.
+
+        The searcher is supposed to maximize reward, while internally, the
+        criterion is minimized. map_reward maps reward to internal criterion, it
+        must be strictly decreasing.
+
+        :param hp_ranges: Configuration space without resource attribute
+        :param random_seed:
+        :param output_gpmodels: Dict with the GP regression model for the active metric and constraint metric
+        :param output_models_args: Dict with arguments for GPMXNet model creation for the active and constraint metric
+        :param map_reward: Function mapping reward to criterion to be minimized
+        :param acquisition_class: Type for acquisition function
+        :param init_state: TuningJobState to start from (default is empty)
+        :param local_minimizer_class: Type for local minimizer
+        :param skip_optimization: Predicate, see
+            GPMXNetPendingCandidateStateTransformer
+        :param num_initial_candidates: See BayesianOptimizationAlgorithm
+        :param num_initial_random_choices: Configs are sampled at random until
+            this many candidates received label feedback
+        :param initial_scoring: Scoring function to rank initial candidates.
+            Default: thompson_indep (independent Thompson sampling)
+        :param first_is_default: If true, the first config to be evaluated is
+            the default of the search space. Otherwise, the first is sampled
+            at random
+        :param debug_log: DebugLogPrinter for debug logging (optional)
+        :param cost_metric_name: If an `elapsed_time` record is passed to
+            `update`, the value is entered in the state using this key
+        :param profiler: If given, HPO computations are profiled
+        :param getconfig_callback: If given, this callback function is called
+            at the end of each get_config. It receives the chosen config
+            (as dict) as single argument. One use case for this callback is
+            to store profiler records.
+        """
+        self.active_metric_name = DEFAULT_METRIC
```

Comment:
These two seem to be parameters the user can pass elsewhere, e.g. in searcher_factory.py.

I think either way is OK (user can choose, or we fix them). Since these are internal names, it is fine to fix them. But please only do one of them.

Reply:
I've now fixed them as they are internal only.